File: 2.sauvegarder_zip.go
package main

import (
	"archive/zip"
	"fmt"
	"io"
	"os"
	"path/filepath"
)

func main() {
	sourceDir := `C:\Users\Public\TP_Automatisation\test_zip`
	os.MkdirAll(sourceDir, os.ModePerm)

	// créer fichiers test
	for i := 0; i < 3; i++ {
		f, _ := os.Create(filepath.Join(sourceDir, fmt.Sprintf("file%d.txt", i)))
		f.WriteString("test\n")
		f.Close()
	}

	zipFile := `C:\Users\Public\TP_Automatisation\backup.zip`
	zf, _ := os.Create(zipFile)
	defer zf.Close()
	zipWriter := zip.NewWriter(zf)
	defer zipWriter.Close()

	filepath.Walk(sourceDir, func(path string, info os.FileInfo, err error) error {
		if !info.IsDir() {
			f, _ := os.Open(path)
			defer f.Close()
			w, _ := zipWriter.Create(info.Name())
			io.Copy(w, f)
		}
		return nil
	})
	fmt.Println("Archive créée :", zipFile)
}

File: 2.sauvegarder_zip.py
import shutil
import os

source_dir = r"C:\Users\Public\TP_Automatisation\test_zip"
os.makedirs(source_dir, exist_ok=True)

# créer des fichiers de test
for i in range(3):
    with open(os.path.join(source_dir, f"file{i}.txt"), "w") as f:
        f.write("test\n")

backup_file = r"C:\Users\Public\TP_Automatisation\backup"
shutil.make_archive(backup_file, "zip", source_dir)
print(f"Archive créée : {backup_file}.zip")
